### DISCUSSION TITLE
How can we concatenate multiple env vars at workflow and job level?

### DISCUSSION THREAD

**Discussion by adamrushuk** [community/community#26179] [Actions]
<p>I’m trying to concatenate a Workflow env var (PREFIX) with other Workflow and Job env vars, but it looks like I can only concatenate within the Step scope.</p>
<p>Using Jenkins I could do this:</p>
<pre><code>environment {
    PREFIX = "rush"
    ACR_FQDN = "${PREFIX}.azurecr.io"
}
</code></pre>
<p>Here is my failing example Workflow:</p>
<pre><code>name: test

on: [push]

env:
  # this works
  MY_WORKFLOW_VAR: work
  # this does NOT work
  MY_OTHER_WORKFLOW_VAR: ${{ env.MY_WORKFLOW_VAR }}-value01

jobs:
  build:

    runs-on: windows-2019

    env:
      # this works
      MY_JOB_VAR: job
      # this does NOT work
      MY_WORK_JOB_VAR: ${{ env.MY_WORKFLOW_VAR }}-somejob01

    steps:
    - name: Output event data
      env:
        MY_STEP_VAR: step
        # this works
        MY_OTHER_STEP_VAR: ${{ env.MY_WORKFLOW_VAR }}-somevalue01
      run: |
        # Env var concatenation
        Write-Output "MY_WORKFLOW_VAR is: [$env:MY_WORKFLOW_VAR]"
        Write-Output "MY_JOB_VAR is: [$env:MY_JOB_VAR]"
        Write-Output "MY_STEP_VAR is: [$env:MY_STEP_VAR]"
        Write-Output "MY_WORK_JOB_VAR is: [$env:MY_WORK_JOB_VAR]"
        Write-Output "MY_OTHER_STEP_VAR is: [$env:MY_OTHER_STEP_VAR]"
</code></pre>
<p>Currently, if I want to use concatenated env vars across multiple steps, I have to repeat the Step level env var definitions like so:</p>
<pre><code>- name: Step 1
      run: pwsh -command "./scripts/step1.ps1"
      env:
        MY_STEP_VAR: ${{ env.MY_WORKFLOW_VAR }}-001
        MY_OTHER_STEP_VAR: ${{ env.MY_OTHER_WORKFLOW_VAR }}-002

    - name: Step 2
      run: pwsh -command "./scripts/step1.ps1"
      env:
        MY_STEP_VAR: ${{ env.MY_WORKFLOW_VAR }}-001
        MY_OTHER_STEP_VAR: ${{ env.MY_OTHER_WORKFLOW_VAR }}-002
</code></pre>
<p>Is there a way to concatenate env vars higher up, in Workflow or Job scope?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/adamrushuk">@adamrushuk</a> ,</p>
<p>The worklfow&amp;job level env doesn’t support variable expression, hence cannot concatenate other workflow/job env vars. Currently it’s only supported for step level env.</p>
<p>You don’t need to repeat step level env definition, you can use ‘<a href="https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-an-environment-variable-set-env" rel="nofollow noopener">set-env</a>’ to create and update env variable for any actions running next in a job. Code sample as below:</p>
<pre><code>name: my workflow
on: push
env:
  MY_WORKFLOW_VAR: work
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Set env
      run: echo ::set-env name=STEPVAR::${{ env.MY_WORKFLOW_VAR }}-somestep
    - name: Test
      run: echo $STEPVAR
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181095941-bd26b0e2-777d-4a55-a7bd-1246310d2c44.png" alt="envsetting.png" data-base62-sha1="ow7x5VEZheOFCG0RL8KIggA4XHu" width="432" height="270"></p>
<p>Thanks.</p>

**Comment by adamrushuk** [comment]
<p>Thanks for the workaround weide-zhou; this works perfectly!</p>
